### PR TITLE
Implements `PagePromotional`s

### PR DESCRIPTION
**Pull request by colinta**
If the `Category` handed to `CategoryViewController` is a `level == .Meta` category, the `page_promotionals` endpoint will be fetched, and a random promotional will be selected as the "primary jsonable" for the category.

New cell type `PagePromotionalHeader`, and renamed `CategoryHeader` to `CategoryPromotionalHeader`.

### CODE REVIEW

**Comment by colinta** [comment]
@steam This could wait until specs are ready, and be rebased then.  At that point I'd like to rename `CategoryHeaderCellPresenter` to `CategoryPromotionalHeaderCellPresenter`.

File: Sources/Controllers/Categories/CategoryGenerator.swift [hunk 1]
```diff
@@ -8,7 +8,11 @@ public final class CategoryGenerator: StreamGenerator {
     public var streamKind: StreamKind
     weak public var destination: StreamDestination?
 
-    private var category: Category?
+    private var category: Category
+    private var categoryIsMeta: Bool {
```
Comment (steam) [inline Sources/Controllers/Categories/CategoryGenerator.swift]: We could move this to `Category.swift` and use it here as well as in `CategoryViewController.swift`.

File: Sources/Controllers/Categories/CategoryViewController.swift [hunk 2]
```diff
@@ -133,17 +137,21 @@ extension CategoryViewController: StreamDestination {
...
+            else {
+                categoryPromotional = category.randomPromotional
+            }
+
+            self.title = category.name
         }
-        else {
-            categoryPromotional = category.randomPromotional
+        else if let pagePromotional = jsonable as? PagePromotional {
+            self.pagePromotional = pagePromotional
```
Comment (steam) [inline Sources/Controllers/Categories/CategoryViewController.swift]: Do we have the same potential issue of pull-to-refresh image swapping with page promotionals that we fixed for categoryPromotionals above?

Comment (colinta) [inline Sources/Controllers/Categories/CategoryViewController.swift]: Actually, the promotional is stored in the generator (along with the category).  I added it here for consistency only (it's not used anywhere in the controller).  On reload a new promotional will be displayed.

File: Sources/Controllers/Categories/CategoryGenerator.swift (hunk 1 above)
Comment (colinta) [inline Sources/Controllers/Categories/CategoryGenerator.swift]: I like it!

File: Sources/Controllers/Categories/CategoryViewController.swift (hunk 2 above)
Comment (steam) [inline Sources/Controllers/Categories/CategoryViewController.swift]: 👍

**Comment by steam** [comment]
@colinta - good to merge?

**Pull request merged**